### PR TITLE
Make shared boost the default, simpler VERSION_STRING, fix Doxygen

### PR DESCRIPTION
**Pull request by vespakoen**
Let's see what CI says ;)

### CODE REVIEW

**Comment by petterreinholdtsen** [comment]
Well, what was learned from this?

**Comment by petterreinholdtsen** [comment]
[Koen Schmeets]
> There is still some work to do on this. I am working on it in my free
> time sporadically, might take a while ;)

I guess we all are doing this in our spare time. :)

Anything I an do to help?  I would like to get a version into Debian
with less local patches, as it will require less effort for me to
maintain it in the future. :)

-- 
Happy hacking
Petter Reinholdtsen

**Comment by petterreinholdtsen** [comment]
Hi.  Any news on a new release?

he library is now installed on 0.13% of the Debian population, at least
285 machines. See "popcon" link on
<URL: https://tracker.debian.org/pkg/opencamlib > for the amazing
graph. :)

-- 
Happy hacking
Petter Reinholdtsen

**Comment by vespakoen** [comment]
I am having issues with the Windows version, using boost python precompiled from boost.org.

See: https://github.com/aewallin/opencamlib/actions/runs/6157111279/job/16707179864

It seems to be linking to dll's that are later not found anymore.

I setup a Windows VM on my own machine and got it to compile with no problems, I am not sure what is going wrong in CI and it is hard to figure out what it has linked to, since the VM doesn't have "dumpbin" and I am not sure how to install it.

I think after all this hassle I gave up on this for a bit.

So in short, it looks like everything is working (CD workflow is fine, using statically compiled boost)
Shared boost (in Test workflow) works on macOS and Linux, and also on my local Windows VM, just not in Github CI.

I would like to figure out why first...
